MfciPkg: MfciPolicyParsingLib Remove Debug statement.

apop5 opened the pull request:
## Description

FindRule used a debug statement to attempt to output a string from the passed policy binary. String statements in the policy binary are not null terminated, becuase the string legnth is a different parameter. The debug statement would access beyond the end of the policy string attempting to find a null termination, and at least in the MfciPolicyParsingUnitTest would access an unmapped page of memory, resulting in a fault/hang.

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?
- [X] Backport to release branch?

## How This Was Tested
Running Unit test on Q35 would results in a hang/page fault.

## Integration Instructions
No integration necessary.

apop5 committed:
MfciPkg: MfciPolicyParsingLib Remove Debug statement.

FindRule used a debug statement to attempt to output a string from the
passed policy binary. String statements in the policy binary are not null
terminated, becuase the string legnth is a different parameter. The debug
statement would access beyond the end of the policy string attempting
to find a null termination, and at least in the MfciPolicyParsingUnitTest
would access an unmapped page of memory, resulting in a fault/hang.

makubacki commented:
@apop5, there are some other changes in the function I think would be nice to have, but I can't comment there because of GitHub comment restrictions. Would you be okay with me pushing a commit to your PR branch? Or would you rather that be a follow up PR?

apop5 commented:
> @apop5, there are some other changes in the function I think would be nice to have, but I can't comment there because of GitHub comment restrictions. Would you be okay with me pushing a commit to your PR branch? Or would you rather that be a follow up PR?

Yes, please feel free to push improvements

makubacki committed:
MfciPolicyParsingLib - FindRule(): Prevent another deref issue in DEBUG()

Print the input string after it has been checked with the safe string
checks in `StrCpyS()` before printing in the debug message.

Miscellaneious tweaks to printing integers in other nearby prints.

Signed-off-by: Michael Kubacki <michael.kubacki@microsoft.com>

makubacki commented:
> Yes, please feel free to push improvements

Thanks. Pushed.

apop5 commented:
Ran Q35 202502 unit tests with the latest changes and everything is passing.
